Documentation/Contributors/CodingGuide/README.md

```diff
@@ -0,0 +1,662 @@
+
+if (!defined(result)) {
+   // ...
+}
+```
+* Use parenthesis judiciously, e.g.,
+```javascript
+var foo = x > 0.0 && y !== 0.0;
+```
+is better written as
+```javascript
+var foo = (x > 0.0) && (y !== 0.0);
+```
+* Use vertical whitespace to separate functions and to group related statements within a function, e.g.,
+```javascript
+var Model = function(options) {
+    // ...
+    this._allowPicking = defaultValue(options.allowPicking, true);
+
+    this._ready = false;
+    this._readyPromise = when.defer();
+    // ...
+};
+```
+* Use single quotes, `'`, instead of double quotes, `"`.  `"use strict"` is an exception and should use double quotes.
+
+## Units
+
+* Use meters for distances.
+* Use radians for angles, except for explicit `Degrees` functions such as `Cartesian3.fromDegrees`.
+* Use seconds for time durations.
+
+## Basic Code Construction
+
+* To avoid type coercion, test for equality with `===` and `!==`, e.g.,
+```javascript
+var f = 1.0;
+
+if (f === 1.0) {
+    // ...
+}
+
+if (f !== 1.0) {
+    // ...
+}
+```
+* To aid the human reader, append `.0` to whole numbers intended to be floating-point values, e.g., unless `f` is an integer,
+```javascript
+var f = 1;
+```
+is better written as
+```javascript
+var f = 1.0;
+```
+* Declare variables where they are first used.  For example,
+```javascript
+var i;
+var m;
+var models = [ /* ... */ ];
+var length = models.length;
+for (i = 0; i < length; ++i) {
+    m = models[i];
+    // Use m
+}
+```
+is better written as
+```javascript
+var models = [ /* ... */ ];
+var length = models.length;
+for (var i = 0; i < length; ++i) {
+    var m = models[i];
+    // Use m
+}
+```
+* Variables have function-level, not block-level scope.  Do not rely on variable hoisting, i.e., using a variable before it is declared, e.g.,
+```javascript
+console.log(i); // i is undefined here.  Never use a variable before it is declared.
+var i = 0.0;
+```
+* Use `undefined` instead of `null`.
+* Test if a variable is defined using Cesium's `defined` function, e.g.,
+```javascript
+var v = undefined;
+if (defined(v)) {
+    // false
+}
+
+var u = {};
+if (defined(u)) {
+    // true
+}
+```
+* Use Cesium's `definedNotNull` function to test the return of `String.match()` and JSON properties.
+* Use Cesium's `freezeObject` function to create enums, e.g.,
+```javascript
+/*global define*/
+define([
+        '../Core/freezeObject'
+    ], function(
+        freezeObject) {
+    "use strict";
+
+    var ModelAnimationState = {
+        STOPPED : 0,
+        ANIMATING : 1
+    };
+
+    return freezeObject(ModelAnimationState);
+});
+```
+* Use descriptive comments for non-obvious code, e.g.,
+```javascript
+byteOffset += sizeOfUint32; // Add 4 to byteOffset
+```
+is better written as
+```javascript
+byteOffset += sizeOfUint32; // Skip length field
+```
+* `TODO` comments need to be removed or addressed before the code is merged into master.  Used sparingly, `PERFORMANCE_IDEA`, can be handy later when profiling.
+
```

Comment:
You might want to add that we do not submitted commented out code.  If a section of code is no longer needed, delete it.  This is unfortunately still a common practice for some people.